src/library/scala/collection/GenTraversableOnce.scala

```diff
@@ -67,6 +67,20 @@ trait GenTraversableOnce[+A] extends Any {
     */
   def foreach[U](f: A => U): Unit
 
+  /** Tests whether this $coll is known to have a finite size.
+    *  All strict collections are known to have finite size. For a non-strict
+    *  collection such as `Stream`, the predicate returns `'''true'''` if all
+    *  elements have been computed. It returns `'''false'''` if the stream is
+    *  not yet evaluated to the end. Non-empty Iterators usually return
+    *  `'''false'''` even if they were created from a collection with a known
+    *  finite size.
+    *
+    *  Note: many collection methods will not work on collections of infinite sizes
+    *  (but they do not require this method to return `'''true'''`).
```

Comment:
Yes, that longer explanation looks good.  It clearly and fairly succinctly explains the issues without assuming mastery of the paragraph that was presumably just read a few seconds before.  Maybe the last sentence would be clearer, "However, checking `hasDefiniteSize` can provide an assurance that `size` is well-defined and non-termination is not a concern."  (Because it's not actually the `hasDefiniteSize == true` that _causes_ the termination of the traversal.  It's just a promise.)